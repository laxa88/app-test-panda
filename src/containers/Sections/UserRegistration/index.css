.leftSection {
  background-color: green;
}

.rightSection {
  background-color: green;
}

.optionsSection {
  background-color: aqua;
}

.buttonsSection {
  background-color: pink;
}