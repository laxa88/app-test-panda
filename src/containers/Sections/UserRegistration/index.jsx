import PropTypes from 'prop-types';
import React from 'react';

import Label from '../../../components/Label';
import InputWithInstructions from '../../../components/InputWithInstructions';
import RadioGroup from '../../../components/RadioGroup';
import CheckboxGroup from '../../../components/CheckboxGroup';
import ButtonPrimary from '../../../components/Button/ButtonPrimary';
import ButtonSecondary from '../../../components/Button/ButtonSecondary';
import css from './index.css';

const rideInGroupOptions = [
  { value: 0, label: 'Always' },
  { value: 1, label: 'Sometimes' },
  { value: 2, label: 'Never' },
];

const daysOfTheWeekOptions = [
  { value: 0, label: 'Sun' },
  { value: 1, label: 'Mon' },
  { value: 2, label: 'Tue' },
  { value: 3, label: 'Wed' },
  { value: 4, label: 'Thu' },
  { value: 5, label: 'Fri' },
  { value: 6, label: 'Sat' },
];

class UserRegistration extends React.Component {
  state = {
    fullName: '',
    email: '',
    city: '',
    rideInGroupIndex: null,
    daysOfTheWeekIndices: [],
  };

  handleOnChangeFullName = (e) => {
    this.setState({ fullName: e.currentTarget.value });
  };

  handleOnChangeEmail = (e) => {
    this.setState({ email: e.currentTarget.value });
  };

  handleOnChangeCity = (e) => {
    this.setState({ city: e.currentTarget.value });
  };

  handleOnChangeRideInGroup = ({ index }) => {
    this.setState({ rideInGroupIndex: index });
  };

  handleOnChangeDaysOfTheWeek = ({ checkedIndices }) => {
    this.setState({ daysOfTheWeekIndices: checkedIndices });
  };

  handleOnClickCancel = () => {
    const { onCancel } = this.props;

    onCancel();
  };

  handleOnClickSave = () => {
    const { onSave } = this.props;

    const {
      fullName,
      email,
      city,
      rideInGroupIndex,
      daysOfTheWeekIndices,
    } = this.state;

    const payload = {
      fullName,
      email,
      city,
      rideInGroup: rideInGroupOptions[rideInGroupIndex],
      daysOfTheWeek: daysOfTheWeekOptions.map((item, index) => ({
        value: item.value,
        label: item.label,
        checked: !!daysOfTheWeekIndices[index],
      })),
    };

    onSave(payload);
  };

  render() {
    const {
      fullName,
      email,
      city,
      rideInGroupIndex,
      daysOfTheWeekIndices,
    } = this.state;

    return (
      <div>
        <h1>User Registration</h1>

        <div className={css.leftSection}>
          <Label text="Full name">
            <InputWithInstructions
              value={fullName}
              onChange={this.handleOnChangeFullName}
              instructions="Enter your full name here"
            />
          </Label>

          <Label text="E-mail">
            <InputWithInstructions
              value={email}
              onChange={this.handleOnChangeEmail}
              instructions="Enter your email here"
            />
          </Label>

          <Label text="City">
            <InputWithInstructions
              value={city}
              onChange={this.handleOnChangeCity}
              instructions="Enter your city here"
            />
          </Label>
        </div>

        <div className={css.rightSection}>
          <div className={css.optionsSection}>
            <Label text="Ride in a group?">
              <RadioGroup
                checkedIndex={rideInGroupIndex}
                options={rideInGroupOptions}
                onChange={this.handleOnChangeRideInGroup}
              />
            </Label>

            <Label text="Ride in a group?">
              <CheckboxGroup
                checkedIndices={daysOfTheWeekIndices}
                options={daysOfTheWeekOptions}
                onChange={this.handleOnChangeDaysOfTheWeek}
              />
            </Label>
          </div>

          <div className={css.buttonsSection}>
            <ButtonSecondary onClick={this.handleOnClickCancel}>Cancel</ButtonSecondary>
            <ButtonPrimary onClick={this.handleOnClickSave}>Save</ButtonPrimary>
          </div>
        </div>
      </div>
    );
  }
}

UserRegistration.propTypes = {
  onCancel: PropTypes.func.isRequired,
  onSave: PropTypes.func.isRequired,
};

export default UserRegistration;
