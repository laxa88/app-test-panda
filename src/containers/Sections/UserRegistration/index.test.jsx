import React from 'react';
import renderer from 'react-test-renderer';

import UserRegistration from './index';

jest.mock('moment', () => () => ({
  default: jest.fn().mockReturnThis(),
  format: () => 'dummy format',
}));

describe('<UserRegistration />', () => {
  const getMockProps = () => ({
    onSave: jest.fn(),
  });

  it('renders', () => {
    const mockProps = getMockProps();
    const result = renderer.create(<UserRegistration {...mockProps} />);

    expect(result.toJSON()).toMatchSnapshot();
  });

  it('triggers handleOnChangeFullName', () => {
    const mockProps = getMockProps();
    const result = renderer.create(<UserRegistration {...mockProps} />);

    const mockEvent = {
      currentTarget: {
        value: 'dummy value',
      },
    };

    result.getInstance().handleOnChangeFullName(mockEvent);

    expect(result.getInstance().state.fullName).toBe(mockEvent.currentTarget.value);
  });

  it('triggers handleOnChangeEmail', () => {
    const mockProps = getMockProps();
    const result = renderer.create(<UserRegistration {...mockProps} />);

    const mockEvent = {
      currentTarget: {
        value: 'dummy value',
      },
    };

    result.getInstance().handleOnChangeEmail(mockEvent);

    expect(result.getInstance().state.email).toBe(mockEvent.currentTarget.value);
  });

  it('triggers handleOnChangeCity', () => {
    const mockProps = getMockProps();
    const result = renderer.create(<UserRegistration {...mockProps} />);

    const mockEvent = {
      currentTarget: {
        value: 'dummy value',
      },
    };

    result.getInstance().handleOnChangeCity(mockEvent);

    expect(result.getInstance().state.city).toBe(mockEvent.currentTarget.value);
  });

  it('triggers handleOnChangeRideInGroup', () => {
    const mockProps = getMockProps();
    const result = renderer.create(<UserRegistration {...mockProps} />);

    const mockEvent = {
      index: 123,
    };

    result.getInstance().handleOnChangeRideInGroup(mockEvent);

    expect(result.getInstance().state.rideInGroupIndex).toBe(mockEvent.index);
  });

  it('triggers handleOnChangeDaysOfTheWeek', () => {
    const mockProps = getMockProps();
    const result = renderer.create(<UserRegistration {...mockProps} />);

    const mockEvent = {
      checkedIndices: [true, false, true],
    };

    result.getInstance().handleOnChangeDaysOfTheWeek(mockEvent);

    expect(result.getInstance().state.daysOfTheWeekIndices).toBe(mockEvent.checkedIndices);
  });

  it('triggers handleOnClickCancel', () => {
    const mockProps = getMockProps();
    const result = renderer.create(<UserRegistration {...mockProps} />);

    result.getInstance().handleOnClickCancel();

    mockProps.onCancel = jest.fn();
    result.update(<UserRegistration {...mockProps} />);

    // result.getInstance().props.onCancel = mockCancel;

    result.getInstance().handleOnClickCancel();

    expect(mockProps.onCancel).toHaveBeenCalledTimes(1);
  });

  it('triggers handleOnClickSave', () => {
    const mockProps = getMockProps();

    const result = renderer.create(<UserRegistration {...mockProps} />);

    const mockState = {
      fullName: 'dummy fullName',
      email: 'dummy email',
      city: 'dummy city',
      rideInGroupIndex: 123,
      daysOfTheWeekIndices: [true, false, true],
      registrationDay: 'dummy format',
    };

    result.getInstance().setState(mockState);

    const expected = {
      fullName: mockState.fullName,
      email: mockState.email,
      city: mockState.city,
      rideInGroupIndex: mockState.rideInGroupIndex,
      daysOfTheWeekIndices: mockState.daysOfTheWeekIndices,
      registrationDay: mockState.registrationDay,
    };

    result.getInstance().handleOnClickSave();

    expect(mockProps.onSave).toHaveBeenCalledWith(expected);
  });
});
