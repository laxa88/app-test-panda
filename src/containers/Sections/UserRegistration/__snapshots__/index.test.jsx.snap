// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserRegistration /> renders 1`] = `
<div
  className="container"
>
  <h1
    className="title"
  >
    User Registration
  </h1>
  <div
    className="leftSection"
  >
    <div>
      <div
        className="sc-bdVaJa kWxWaB"
      >
        Full name
      </div>
      <div>
        <input
          className="sc-bwzfXH eIPCFZ"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          value=""
        />
        <div
          className="sc-htpNat fikBex"
          style={
            Object {
              "opacity": 0,
            }
          }
        >
          Enter your full name here
        </div>
      </div>
    </div>
    <div>
      <div
        className="sc-bdVaJa kWxWaB"
      >
        E-mail
      </div>
      <div>
        <input
          className="sc-bwzfXH eIPCFZ"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          value=""
        />
        <div
          className="sc-htpNat fikBex"
          style={
            Object {
              "opacity": 0,
            }
          }
        >
          Enter your email here
        </div>
      </div>
    </div>
    <div>
      <div
        className="sc-bdVaJa kWxWaB"
      >
        City
      </div>
      <div>
        <input
          className="sc-bwzfXH eIPCFZ"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          value=""
        />
        <div
          className="sc-htpNat fikBex"
          style={
            Object {
              "opacity": 0,
            }
          }
        >
          Enter your city here
        </div>
      </div>
    </div>
  </div>
  <div
    className="rightSection"
  >
    <div
      className="row inputSection"
    >
      <div>
        <div
          className="sc-bdVaJa kWxWaB"
        >
          Ride in a group?
        </div>
        <div
          className="sc-gzVnrw hsxAGA"
        >
          <button
            checked={false}
            className="sc-bxivhb gdYtMu"
            disabled={false}
            onClick={[Function]}
          >
            <div
              checked={false}
              className="sc-ifAKCX gVhslb"
            >
              <div
                checked={false}
                className="sc-EHOje eXoKJA"
              />
            </div>
            <div
              className="sc-bZQynM jHMOIg"
            >
              Always
            </div>
          </button>
          <button
            checked={false}
            className="sc-bxivhb gdYtMu"
            disabled={false}
            onClick={[Function]}
          >
            <div
              checked={false}
              className="sc-ifAKCX gVhslb"
            >
              <div
                checked={false}
                className="sc-EHOje eXoKJA"
              />
            </div>
            <div
              className="sc-bZQynM jHMOIg"
            >
              Sometimes
            </div>
          </button>
          <button
            checked={false}
            className="sc-bxivhb gdYtMu"
            disabled={false}
            onClick={[Function]}
          >
            <div
              checked={false}
              className="sc-ifAKCX gVhslb"
            >
              <div
                checked={false}
                className="sc-EHOje eXoKJA"
              />
            </div>
            <div
              className="sc-bZQynM jHMOIg"
            >
              Never
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      className="row inputSection"
    >
      <div
        style={
          Object {
            "padding": "8px 0",
          }
        }
      >
        <div
          className="sc-bdVaJa kWxWaB"
        >
          Days of the week
        </div>
        <div
          className="sc-iwsKbI iNklh"
        >
          <div
            className="sc-gqjmRU liHAQW"
          >
            <button
              checked={false}
              className="sc-htoDjs clIbxB"
              disabled={false}
              onClick={[Function]}
            />
            <div
              className="sc-gZMcBi yKZqO"
            >
              Mon
            </div>
          </div>
          <div
            className="sc-gqjmRU liHAQW"
          >
            <button
              checked={false}
              className="sc-htoDjs clIbxB"
              disabled={false}
              onClick={[Function]}
            />
            <div
              className="sc-gZMcBi yKZqO"
            >
              Tue
            </div>
          </div>
          <div
            className="sc-gqjmRU liHAQW"
          >
            <button
              checked={false}
              className="sc-htoDjs clIbxB"
              disabled={false}
              onClick={[Function]}
            />
            <div
              className="sc-gZMcBi yKZqO"
            >
              Wed
            </div>
          </div>
          <div
            className="sc-gqjmRU liHAQW"
          >
            <button
              checked={false}
              className="sc-htoDjs clIbxB"
              disabled={false}
              onClick={[Function]}
            />
            <div
              className="sc-gZMcBi yKZqO"
            >
              Thu
            </div>
          </div>
          <div
            className="sc-gqjmRU liHAQW"
          >
            <button
              checked={false}
              className="sc-htoDjs clIbxB"
              disabled={false}
              onClick={[Function]}
            />
            <div
              className="sc-gZMcBi yKZqO"
            >
              Fri
            </div>
          </div>
          <div
            className="sc-gqjmRU liHAQW"
          >
            <button
              checked={false}
              className="sc-htoDjs clIbxB"
              disabled={false}
              onClick={[Function]}
            />
            <div
              className="sc-gZMcBi yKZqO"
            >
              Sat
            </div>
          </div>
          <div
            className="sc-gqjmRU liHAQW"
          >
            <button
              checked={false}
              className="sc-htoDjs clIbxB"
              disabled={false}
              onClick={[Function]}
            />
            <div
              className="sc-gZMcBi yKZqO"
            >
              Sun
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row buttonsSection"
    >
      <button
        className="button sc-jTzLTM jlDhZY"
        disabled={true}
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="button sc-VigVT jUXDZe"
        disabled={true}
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
